cpp/src/arrow/adapters/orc/adapter_common.h

```diff
@@ -0,0 +1,111 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "arrow/util/visibility.h"
+
+namespace arrow {
+
+namespace adapters {
+
+namespace orc {
+
+enum WriterId {
+  ORC_JAVA_WRITER = 0,
+  ORC_CPP_WRITER = 1,
+  PRESTO_WRITER = 2,
+  SCRITCHLEY_GO = 3,
+  TRINO_WRITER = 4,
+  UNKNOWN_WRITER = INT32_MAX
+};
+
+enum WriterVersion {
+  WriterVersion_ORIGINAL = 0,
+  WriterVersion_HIVE_8732 = 1,
+  WriterVersion_HIVE_4243 = 2,
+  WriterVersion_HIVE_12055 = 3,
+  WriterVersion_HIVE_13083 = 4,
+  WriterVersion_ORC_101 = 5,
+  WriterVersion_ORC_135 = 6,
+  WriterVersion_ORC_517 = 7,
+  WriterVersion_ORC_203 = 8,
+  WriterVersion_ORC_14 = 9,
+  WriterVersion_MAX = INT32_MAX
+};
+
+enum CompressionKind {
```

Comment:
We already have such definitions in `arrow/util/compression.h`, is it actually useful to redefine this?

Reply:
Ah. Not really. You are right.

Reply:
I ended up returning `Status::Invalid` as an element of `Result<liborc::WriterOptions>` for compressions ORC doesn’t support. Is this the right way to do it?